internal/cmd/organizations/list.go

```diff
@@ -35,36 +34,8 @@ func listOrgsCommand() *cobra.Command {
 				return fmt.Errorf("failed to list organizations: %w", err)
 			}
 
-			// TODO: We should look at abstracting the formatting here into a library
-			//       that can be used by multiple commands needing to offer multiple
-			//       output formats from a command.
-			switch outputFormat {
-			case "yaml":
-				marshaller := &protoyaml.MarshalOptions{
-					Indent: 2,
-				}
-				output, err := marshaller.Marshal(listOrgs)
-				if err != nil {
-					return fmt.Errorf("failed to list organizations: %w", err)
-				}
-				fmt.Print(string(output))
-			case "json":
-				output, err := protojson.Marshal(listOrgs)
-				if err != nil {
-					return fmt.Errorf("failed to list organizations: %w", err)
-				}
-				fmt.Print(string(output))
-			case "table":
-				orgTable := table.New("DISPLAY NAME", "RESOURCE ID")
-				if len(listOrgs.Organizations) == 0 {
-					fmt.Printf("No organizations found")
-				} else {
-					for _, org := range listOrgs.Organizations {
-						orgTable.AddRow(org.DisplayName, org.OrganizationId)
-					}
-				}
-				orgTable.Print()
-			}
+			outTableHeaders, outTableData := getListOrganizationsTableOutputData(listOrgs)
```

Comment:
Curious if anyone can think of a more clever way to handle the table header + rowdata more generically. Otherwise, each command will likely need some kind of generator func, similar to `getListOrganizationsTableOutputData`.

Reply:
I was planning to take a look at what GitHub leverages in their `gh` cli tool. 

See https://github.com/cli/cli/blob/6c5145166003ac6fb952c5c591a6f3bdeea10465/pkg/cmd/repo/list/list.go#L179

Their approach seems to have you build the table headers, rows, then render the table, which I think works well and should address the concern on arbitrary data.

Relevant links:

- https://github.com/cli/cli/blob/6c5145166003ac6fb952c5c591a6f3bdeea10465/internal/tableprinter/table_printer.go
- https://github.com/cli/cli/blob/trunk/internal/tableprinter/table_printer.go

Reply:
Ok, that is pretty similar to what I have implemented here at the moment (with the headers and rows being constructed in a func instead of inline). Scot had opted to use `"github.com/rodaine/table"` instead of our own implementation of a table printer, which I am happy to implement if we want to avoid adding external deps. 

For the `gh` cli, table print appears to be first class (It actually doesn't appear to support other formats from a quick glance). 

Thinking outside of the box:
Its certainly possible for us to write a printer which takes in columns, and maps those columns to fields on the object so we can print only those fields in table rows. That seems like it may be overkill unless we find that writing the row builder funcs becomes cumbersome. It would also require the column names we pass to match struct field names (which we could then transform, eg; `DisplayName` becomes `DISPLAY NAME`

Reply:
(Side note, I just moved the header and row build logic inline, instead of in a func we have to test externally, which reduces the surface area for potential bugs. (Since it seems tacitly agreed upon that doing it inline like `gh` is acceptable)

Reply:
> Scot had opted to use "github.com/rodaine/table" instead of our own implementation of a table printer, which I am happy to implement if we want to avoid adding external deps.

I don't have a strong opinion on using an external lib vs printf with some padding, but do know that rendering can get funky with longer strings and I'm going to assume the table library handles that - I haven't looked though.

> Its certainly possible for us to write a printer which takes in columns, and maps those columns to fields on the object so we can print only those fields in table rows. That seems like it may be overkill unless we find that writing the row builder funcs becomes cumbersome. It would also require the column names we pass to match struct field names (which we could then transform, eg; DisplayName becomes DISPLAY NAME

Yeah, I don't think we need to get very crazy here right now, I'd prefer to keep things simple and adjust as we have more demanding use cases.